### PR TITLE
[proxima-direct-io-blob] fix NPE when rollbacking non initialized writer

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java [hunk 1]
```diff
@@ -44,7 +46,7 @@
 
   @Getter private final AccessorT accessor;
   @Getter private final Context context;
-  private transient BulkAttributeWriter wrap;
+  private transient @Nullable BulkAttributeWriter wrap;
```
Comment (dmvk) [inline direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java:49]: under what circumstances is this set to null?

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java [hunk 2]
```diff
@@ -74,12 +76,12 @@ public void write(StreamElement data, long watermark, CommitCallback statusCallb
 
   @Override
   public void updateWatermark(long watermark) {
-    wrap.updateWatermark(watermark);
+    Optional.ofNullable(wrap).ifPresent(w -> w.updateWatermark(watermark));
```
Comment (dmvk) [inline direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java:79]: these kind of extra allocations in internal methods are really uncessary, and don't add anything to overall readability (I'd say readability is even worse)

The correct construct should be.
```
if (wrap != null) {
  wrap.updateWatermark(watermark)
}
```

Is there ANY advantage of using your approach?

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java (hunk 1 above)
Comment (je-ik) [reply direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java:49]: before call to `init`, which is called in write only

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java (hunk 2 above)
Comment (je-ik) [reply direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java:79]: that is only a matter of taste

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java (hunk 1 above)
Comment (dmvk) [reply direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java:49]: 👍

**Pull request merged**